src/ktopt_planning_context.cpp

```diff
@@ -228,6 +235,53 @@ void KTOptPlanningContext::solve(planning_interface::MotionPlanResponse& res)
   return;
 }
 
+void KTOptPlanningContext::addPathPositionConstraints(KinematicTrajectoryOptimization& trajopt,
+                                                      const MultibodyPlant<double>& plant,
+                                                      const Frame<double>& link_ee_frame,
+                                                      Context<double>& plant_context)
+{
+  // retrieve the motion planning request
+  const auto& req = getMotionPlanRequest();
+
+  // check for path position constraints
+  if (!req.path_constraints.position_constraints.empty())
+  {
+    for (const auto& position_constraint : req.path_constraints.position_constraints)
+    {
+      // Extract the bounding box's center (primitive pose)
+      const auto& primitive_pose = position_constraint.constraint_region.primitive_poses[0];
+      Eigen::Vector3d box_center(primitive_pose.position.x, primitive_pose.position.y, primitive_pose.position.z);
+
+      // Extract dimensions of the bounding box from
+      // constraint_region.primitives Assuming it is a box
+      // (shape_msgs::SolidPrimitive::BOX) and has dimensions in [x, y, z]
+      const auto& primitive = position_constraint.constraint_region.primitives[0];
+      if (primitive.type != shape_msgs::msg::SolidPrimitive::BOX)
+      {
+        continue;
```

Comment:
Worth a warning here? E.g.

"Geometric primitive must be a box. Cannot add position constraint."